### ISSUE TITLE
v4  ReferenceArrayInput too small/not wide

### ISSUE THREAD

**Issue by RWOverdijk** [marmelab/react-admin#7408]
**What you were expecting:**

A readable/wide enough input.

**What happened instead:**

An input that's too small (second field).

<img width="334" alt="image" src="https://user-images.githubusercontent.com/781745/159222674-40822fef-870d-4dc0-890e-3fbedf397ece.png">

**Steps to reproduce:**

https://codesandbox.io/s/serverless-feather-cswwec

**Related code:**

`<ExampleCreate />` inside of App.tsx. Specifically the `<ReferenceArrayInput />` part.

- Repository: https://github.com/RWOverdijk/ra-mui-input-bug/blob/main/src/App.tsx#L23-L25
- Codesandbox: https://codesandbox.io/s/serverless-feather-cswwec

Open the user create form. I realise it's using todos (had to find a functional endpoint) and says permissions. It's weird 😄

**Environment**

* React-admin version: 4.0.0-beta.3
* Last version that did not exhibit the issue (if applicable): I guess v3? The codesandbox for version 3 seems to work.
* React version: 17.0.2
* Browser: Chrome 99.0.4844.51
* Stack trace (in case of a JS error): N/A

**Comment by RWOverdijk** [comment]
Working on reproducing the issue. I found a codesandbox that does appear to work fine, so I'm figuring out what the difference is. In case you do end up reading this: it looks like all of my form fields are small. Not _too_ small, just small.

<img width="333" alt="image" src="https://user-images.githubusercontent.com/781745/159224566-2aa92b9c-e0c7-4ad1-90ac-b38deec61b01.png">

**Comment by RWOverdijk** [comment]
Sorry for the mess I'm making here. 

I made and added a codesandbox (in the description). I just remember opening issues here in the past and getting yelled at so I want to be thorough this time 😄

**Comment by CampingPanda** [comment]
The CodeSandbox link has errors and doesn't work for me so can't say much about your example.
But you can just add a style to your fields and inputs like:
```
<ReferenceInput
    label="Bewerking 4"
    source="augment_code_4"
    reference="settings"
    perPage={100}
    filter={{ real_augments: true }}
    style={{ width: "100%" }}    // or any other valid css like "500px", I have this ReferenceInput encapsulated in a 100% width <div>
    allowEmpty={true}
>
```

**Comment by RWOverdijk** [comment]
The codesandbox has errors but still runs. It just takes a long time to install all the dependencies so it looks broken. Unless you were running incognito where localstorage isn't allowed and the sandbox will fail.

Also, I do realise I can style things myself but that goes against why I like react-admin so much 😄. The out-of-the-box behaviour is what this issue is about specifically.

**Comment by CampingPanda** [comment]
My apologies, never had problems with CodeSandbox on Firefox, but this project does not work on it. 
It works on Chrome though. 👍 

What's weird is that it seems to work fine on the simple CodeSandbox from react-admin:
https://codesandbox.io/s/vigorous-leakey-wncecq?file=/src/comments/CommentCreate.tsx
Both are react-admin v4. 
```
const CommentCreate = (props: any) => (
  <Create {...props}>
    <SimpleForm>
      <TextInput source="name" />
      <ReferenceArrayInput source="permission_ids" reference="posts">
        <SelectArrayInput optionText="title" />
      </ReferenceArrayInput>
    </SimpleForm>
  </Create>
);
```
Not really sure what the difference is, but maybe its useful for you.

**Comment by RWOverdijk** [comment]
Ah good, yeah might be a privacy thing.

But your sandbox won't help. It's not v4, it's v3:

```
sandbox@sse-sandbox-113mvx:/sandbox$ yarn list --pattern react-admin
yarn list v1.22.17
└─ react-admin@3.19.10
```

You can double check by seeing that it's based on the simple-template from master (which is v3). Another clue is that it's using redux still.

It would have been easier to diff and solve if it were the same version 😄 The simple-example sandbox is broken for v4: https://codesandbox.io/s/github/marmelab/react-admin/tree/v4.0.0-beta.3/examples/simple which is why I made my own repo.

So the original issue persists I'm afraid. 😞

**Comment by CampingPanda** [comment]
Are you sure about that?
@djhi said you should hover over the dependencies to see the true add-on version here. And he's works @ Marmelab itself, so I have assumed he is correct. 😄
_Originally posted by @djhi in https://github.com/marmelab/react-admin/issues/7385#issuecomment-1066619249_

https://codesandbox.io/s/serverless-feather-cswwec
Your Sandbox: 
![afbeelding](https://user-images.githubusercontent.com/33489797/159646287-8c74d1ff-2c99-49eb-b53c-2facc0b2ac54.png)

https://codesandbox.io/s/vigorous-leakey-wncecq
My Sandbox:
![afbeelding](https://user-images.githubusercontent.com/33489797/159646543-833b2eab-b513-4145-bb8a-f4b91a307063.png)

So I might be misunderstanding, but I think both are react-admin v4.
Anyhow, my apologies for making a mess of your issue, I will stop now. 👍

**Comment by RWOverdijk** [comment]
Yes. I am sure about that. 😄 And thanks though, I appreciate you thinking about this with me.

**Comment by WiXSL** [comment]
Thanks for reporting this.

**Comment by fzaninotto** [comment]
Fixed by #7421

**Comment by RWOverdijk** [comment]
Thanks! I knew I wasn't crazy 😄
